### PR TITLE
Image qc for backgrounds

### PR DESCRIPTION
**Pull request by imagirom**
This implements image level quality control for the background statistics. While it is working as-is, the overall code quality is not good (I wanted a working version first)

@constantinpape maybe you can have a look and point out some better solutions. We could also just talk about this some time tomorrow.

### CODE REVIEW

**Comment by constantinpape** [comment]
Ok, I don't quite understand what you try to achieve. Let's discuss in person first later.

File: batchlib/analysis/cell_level_analysis.py [hunk 1]
```diff
@@ -283,24 +288,78 @@ def save_all_stats(self, in_file, out_file,
...
+    def folder_to_table_path(folder, identifier):
+        # NOTE, we call this .hdf5 to avoid pattern matching, it's a bit hacky ...
+        table_file_name = os.path.split(folder)[1] + '_table.hdf5'
+        return os.path.join(folder, table_file_name)
+
+    @property
+    def table_out_path(self):
+        return self.folder_to_table_path(self.folder, self.identifier)
+
+    def load_image_outliers(self, input_files):
```
Comment (constantinpape) [inline batchlib/analysis/cell_level_analysis.py:313]: Refactor this into `load_image_outlier_dict` and call it in the methods.

Comment (imagirom) [reply batchlib/analysis/cell_level_analysis.py]: This is addressed now. You can have a look if you want. However I did not figure out the job order / input validation of the image qc job yet.

File: batchlib/analysis/cell_analysis_qc.py [hunk 2]
```diff
@@ -139,6 +139,10 @@ def __init__(self,
                          marker_bg_key=marker_bg_key,
                          **super_kwargs)
 
+    def validate_input(self, path):
```
Comment (constantinpape) [inline batchlib/analysis/cell_analysis_qc.py:142]: Is this is still necessary? If yes, we should fix it!

File: batchlib/analysis/cell_level_analysis.py [hunk 3]
```diff
@@ -604,6 +629,112 @@ def validate_outputs(self, output_files, folder, status, ignore_failed_outputs):
...
+
+        self.channel_keys = actual_channels_to_use
+        self.image_outlier_table = image_outlier_table
+
+        group_name = 'backgrounds' if identifier is None else f'backgrounds_{identifier}'
+        self.image_table_key = 'images/' + group_name
+        self.well_table_key = 'wells/' + group_name
+        self.plate_table_key = 'plate/' + group_name
+
+        super(ExtractBackground, self).__init__(
```
Comment (constantinpape) [inline batchlib/analysis/cell_level_analysis.py:658]: I prefer new style super a lot, which is just `super().__Init__(...)`. Could you change it?

File: batchlib/analysis/cell_level_analysis.py [hunk 4]
```diff
@@ -604,6 +629,112 @@ def validate_outputs(self, output_files, folder, status, ignore_failed_outputs):
...
+    assert column_names[0] == 'image_name'
+    column_names = list(column_names)
+    column_names.insert(1, 'site_name')
+    table = np.array(table).tolist()
+    for row in table:
+        row.insert(1, image_name_to_site_name(row[0]))
+    return column_names, np.array(table)
+
+
+class ExtractBackground(CellLevelAnalysisWithTableBase):
```
Comment (constantinpape) [inline batchlib/analysis/cell_level_analysis.py:642]: The `cell_level_analysis.py` file is getting a bit long, should we maybe refactor `InstanceFeatureExtraction` and this class to a new file, maybe `cell_level_features.py`?

Comment (imagirom) [reply batchlib/analysis/cell_level_analysis.py:624]: Yes, we should do that at some point soon.

File: batchlib/base.py [hunk 5]
```diff
@@ -7,8 +7,10 @@
 from .config import get_default_extension
 from .util import is_group, get_file_lock, get_logger
 from .util import io as io
+import logging
 
 logger = get_logger('Workflow.BatchJob')
+logger.setLevel(logging.DEBUG)
```
[the remark this answers is not in the document]
Comment (imagirom) [reply batchlib/base.py:13]: Whoops, commited on accident..

File: batchlib/workflows/cell_analysis.py [hunk 6]
```diff
@@ -167,11 +186,11 @@ def run_cell_analysis(config):
...
             # new method
-            'bg_correction_key': 'well_bg_median',
-            'infected_threshold_scale_key': 'well_bg_mad',
-            'infected_threshold': 7,
+            'bg_correction_key': 'plate/backgrounds',
+            'scale_with_mad': True,
+            'infected_threshold': 6,
         },
-        'run': {'force_recompute': None}}))
+        'run': {'force_recompute': True}}))
```
[the remark this answers is not in the document]
Comment (imagirom) [reply batchlib/workflows/cell_analysis.py:193]: Same as above

**Comment by constantinpape** [comment]
Ok, I will merge as soon as the tests pass.

File: batchlib/analysis/cell_analysis_qc.py (hunk 2 above)
Comment (imagirom) [reply batchlib/analysis/cell_analysis_qc.py:139]: I will look at it now

**Pull request merged**